pkg/kv/store.go

```diff
@@ -125,6 +125,12 @@ type EntriesIterator interface {
 	Close()
 }
 
+type ResultIterator interface {
+	EntriesIterator
+	IsInRange(key []byte) bool
```

Comment:
This doesn't feel natural to me. `Range` isn't part of the interface. It's a private implementation detail of some stores (but not others) and it's like hinting the caller how to help the iterator to perform well. An alternative would be that the iterator would perform well without hints being passed. For example -> `SeekGE` should check its entries in memory before making another request to the DB. I feel like this makes more sense, not sure how feasible it really is.